node/src/main/scala/coop/rchain/node/api/DeployGrpcService.scala

```diff
@@ -67,7 +67,20 @@ private[api] class DeployGrpcService[
 
   override def createBlock(e: Empty): Future[MaybeBlockMessage] = BlockAPI.createBlock[F].toFuture
 
-  override def addBlock(b: BlockMessage): Future[Empty] = BlockAPI.addBlock[F](b).toFuture
```

Comment:
I think it would be better to update `BlockAPI.addBlock` with the new logic and leave this file unchanged.

Reply:
should I also move the logic for `doDeploy` to `BlockApi`?

Reply:
Yes, that could be a good think to do for consistency. `BlockAPI` did not exist when `doDeploy` was first written.